core/src/main/scala/org/apache/spark/storage/DiskBlockManager.scala

```diff
@@ -139,8 +139,8 @@ private[spark] class DiskBlockManager(blockManager: BlockManager, conf: SparkCon
   }
 
   private def addShutdownHook(): AnyRef = {
-    Utils.addShutdownHook { () =>
-      logDebug("Shutdown hook called")
+    Utils.addShutdownHook(Utils.TEMP_DIR_SHUTDOWN_PRIORITY + 1) { () =>
```

Comment:
Minor: but should we just add another constant for this in the main file? It might be good to have all the relevant constants there, since ultimately this is a problem that requires reasoning about global ordering.